pkg/bundle/internal/target/target.go

```diff
@@ -396,3 +417,48 @@ func (b *Data) Populate(pool *util.CertPool, formats *trustapi.AdditionalFormats
 	}
 	return nil
 }
+
+// Calculate hash from the given password string.
+func calculateHash(passwd string) string {
+	crcTable := crc32.MakeTable(128)
+	crc := crc32.New(crcTable)
+	crc.Write([]byte(passwd))
+
+	return strconv.Itoa(int(crc.Sum32()))
+}
+
+// Retrieve old password hashes from the given annotations
+// Annotation keys are taken from bundle additional formats keys
+func getOldHashAnnotations(annotations map[string]string, bundle *trustapi.AdditionalFormats) map[string]string {
```

Comment:
I don't think this helper adds any value. If you just loop through the desired annotations in `needsUpdate` and check the desired annotations against the target resource annotations I think the code is easier to read.